lib/rubocop/cop/layout/space_around_operators.rb

```diff
@@ -115,35 +113,36 @@ def on_send(node)
         end
 
         def on_assignment(node)
-          _, rhs, = *node
+          rhs = node.rhs
 
           return unless rhs
 
           check_operator(:assignment, node.loc.operator, rhs)
         end
 
-        def on_casgn(node)
-          _, _, right, = *node
+        def on_class(node)
+          rhs = node.parent_class
 
-          return unless right
+          return unless rhs
 
-          check_operator(:assignment, node.loc.operator, right)
+          check_operator(:class, node.loc.operator, rhs)
         end
 
         def on_binary(node)
-          _, rhs, = *node
+          rhs = node.rhs
 
           return unless rhs
 
           check_operator(:binary, node.loc.operator, rhs)
         end
 
         def on_special_asgn(node)
-          _, _, right, = *node
+          # node.first_argument - is this correct?
+          rhs = node.send_type? ? node.first_argument : node.rhs
```

Comment:
@dvandersluis Is this correct? When do I use `SendNode#first_argument` vs `SendNode#last_argument`?

Reply:
In this case yes it's correct because it's not possible to pass multiple arguments to a setter method. `x.y = 3, 5` acts as an implicit array (`[3,5]`), so it'll always be the first argument there. Probably would be good to add this to the tests of this cop.

FWIW I'd probably refactor `on_special_asgn` to `on_setter_method` to only deal with the send node case, and then the other case can be handled by `on_assignment` as well since it'll be the same at that point except for the type, which could be set in `on_assignment`.

Reply:
I didn’t quite follow that. Would you be up for pushing a commit to my branch?

Reply:
@bquorning I pushed a commit with what I meant.